components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/GeneListController.java

```diff
@@ -246,45 +258,96 @@ public void writeJSON(Patient patient, JSONObject json, Collection<String> selec
 
         try {
             JSONArray genesJson = json.getJSONArray(this.getJsonPropertyName());
```

Comment:
This may be missing from the 1.2 JSON, so it should also be retrieved with `optJSONArray`, and `parseGenesJson` should start with an `if not null`.